CMakeLists.txt

```diff
@@ -1,342 +1,186 @@
+# Part of CMake configuration for GEOS
 #
-# Copyright (C) 2009 Mateusz Loskot <mateusz@loskot.net>
+# Copyright (C) 2018 Mateusz Loskot <mateusz@loskot.net>
 #
 # This is free software; you can redistribute and/or modify it under
 # the terms of the GNU Lesser General Public Licence as published
 # by the Free Software Foundation.
 # See the COPYING file for more information.
-#
-#################################################################################
-cmake_minimum_required(VERSION 3.1.3)
-
-if(NOT CMAKE_VERSION)
-    set(CMAKE_VERSION
-      "${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION}")
-endif()
+################################################################################
+cmake_minimum_required(VERSION 3.5)
 
-if(POLICY CMP0048)
-    cmake_policy(SET CMP0048 NEW)
+# TODO: Follow CMake detection of git and version tagging
+#       https://gitlab.kitware.com/cmake/cmake/blob/master/Source/CMakeVersionSource.cmake
+if(EXISTS ${CMAKE_SOURCE_DIR}/.git/HEAD)
+  set(GEOS_BUILD_FROM_GIT ON)
 endif()
-#################################################################################
-# Set GEOS project
-#################################################################################
-project(GEOS VERSION 3.8.0 LANGUAGES C CXX)
-# OPTIONS: "", "dev", "rc1" etc.
-SET(GEOS_PATCH_WORD "dev")
-
-# Add custom GEOS modules for CMake
-list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/modules")
-
-#################################################################################
-# Set C++ standard
-#################################################################################
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
-
-message(STATUS "Setting C++ requirement to C++${CMAKE_CXX_STANDARD}")
 
-#################################################################################
-# Setup GEOS version
-#################################################################################
+################################################################################
+# Options
+include(CMakeDependentOption)
+## CMake global variables
+option(BUILD_SHARED_LIBS "Build GEOS with shared libraries" OFF)
+## GEOS custom variables
+cmake_dependent_option(GEOS_BUILD_DEVELOPER
+  "Build with compilation flags useful for development" ON
+  "GEOS_BUILD_FROM_GIT" OFF)
+mark_as_advanced(GEOS_BUILD_DEVELOPER)
+#TODO: option(GEOS_BUILD_OSX_FRAMEWORK "Build shared or static library as Framework Bundle for OSX" OFF)
+#mark_as_advanced(GEOS_BUILD_OSX_FRAMEWORK)
 
-# GEOS release version
-# GEOS C++ library SONAME will use these encoding ABI break at every release
-set(GEOS_VERSION "${GEOS_VERSION_MAJOR}.${GEOS_VERSION_MINOR}.${GEOS_VERSION_PATCH}${GEOS_PATCH_WORD}")
-set(VERSION "${GEOS_VERSION}")
+################################################################################
+# Project settings
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.12)
```

Comment:
Looks like `VERSION_GREATER_EQUAL` was introduced in 3.7.

Reply:
Good catch! I'll fix it.